app/javascript/src/components/PatientDashboardForm.jsx

```diff
@@ -26,7 +26,6 @@ export default PatientDashboardForm = ({
 
   const autosave = async (updatedData) => {
     const updatedPatientData = { ...patientData, ...updatedData }
-    setPatientData(updatedPatientData)
```

Comment:
this is the main culprit right?

Reply:
actually the change below with the debounce was the main culprit. a combo of the useMemo and not passing the params through to the autosave function

Reply:
ah, good to know! makes sense

Reply:
I really need to learn to just stop reaching for useMemo. I mess it up nearly every time and it has cost me hours of my life 😅